api/v1beta1/types.go

```diff
@@ -1005,6 +1005,21 @@ type AzureBastion struct {
 	EnableTunneling bool `json:"enableTunneling,omitempty"`
 }
 
+// FleetManager defines the fleet manager configuration.
+type FleetManager struct {
+	// Group is the group this member belongs to for multi-cluster update management.
+	// +optional
+	Group string `json:"group,omitempty"`
+
+	// ManagerName is the name of the fleet manager.
+	// +kubebuilder:validation:Required
+	ManagerName string `json:"managerName,omitempty"`
```

Comment:
`Name` should be enough. From what I can tell no other spec in this file sutter the type for the resource name.
same for `ManagerResourceGroup` below?

Reply:
I just realized that this is a `FleetManagerResourceReference` and not the FleetManager itself.
Would it make sense to rename it this way, to match ClusterResourceReference?

Reply:
The name of the `FleetMember` is not specified the user, as it is set to the `ClusterName` [here](https://github.com/willie-yao/cluster-api-provider-azure/blob/aks-fleets-member/azure/scope/managedcontrolplane.go#L302). I think naming it the `ManagerName` would make it clearer as it is the name of the `FleetManager` we're joining this cluster as a member to, if that makes sense. Also the resource group is specified since the user will be creating the `FleetManager` independently of this cluster.

Reply:
2 member resource cannot have the same name, as the name is the last part that differentiate the resouce ID in azure.

Did you try adding 2 members with this implementation?

Reply:
Yup I have. The member resources are named after the cluster itself and the cluster corresponds 1:1 to a member. Let me double check just to be sure.